bcolz/carray_ext.pyx

```diff
@@ -1386,7 +1386,9 @@ cdef class carray:
 
             # Object dtype requires special storage
             if arrcpy.dtype.char == 'O':
-                self._store_obj(array)
+                n = len(arrcpy)
+                for j in range(n):
```

Comment:
Very minor, but you can probably just write:

```
for j in range(len(arrcpy)):
```

with no need to declare `n` and having the same speed.